src/main/java/htsjdk/tribble/readers/PositionalBufferedStream.java

```diff
@@ -28,7 +28,11 @@
 
 /**
  * A wrapper around an {@code InputStream} which performs it's own buffering, and keeps track of the position.
- * 
+ *
+ * TODO: This class implements Positional, which in turn extends LocationAware, which requires preservation of
```

Comment:
I agree that would be the right thing to do, but htsjdk is kind of painted into a corner at this point. There are several other code paths that would have to change, but worse, BinaryFeatureCodec (and thus BCF2Codec) use PositionalBufferedStream for a SOURCE type parameter, even though BCF can be block compressed. So that codec actually depends on being able to wrap a PositionalBufferedStream around a block compressed stream.